contentcuration/contentcuration/views/base.py

```diff
@@ -188,19 +188,31 @@ def accounts(request):
 )
 @permission_classes((IsAuthenticated,))
 def channel(request, channel_id):
-    channel = get_object_or_404(Channel, id=channel_id, deleted=False)
+    channel_error = ''
 
-    # Check user has permission to view channel
+    # Check if channel exists
     try:
-        request.user.can_view_channel(channel)
-    except PermissionDenied:
-        return HttpResponseNotFound("Channel not found")
+        channel = Channel.objects.get(id=channel_id, deleted=False)
+        channel_id = channel.id
+    except Channel.DoesNotExist:
+        channel_error = 'CHANNEL_EDIT_ERROR_CHANNEL_NOT_FOUND'
+        channel_id = ''
+
+    # Check if user has permission to view channel
+    if channel_id != '':
+        try:
+            request.user.can_view_channel(channel)
+        except PermissionDenied:
+            channel_error = 'CHANNEL_EDIT_ERROR_CHANNEL_NOT_VIEWABLE'
```

Comment:
We might want to treat not found and not viewable both as "not found" errors so people can't find channel ids (especially since we don't check this permission to import channels into Kolibri)

Reply:
I've changed this to handle `PermissionDenied` errors the same as `NotFOund` errors, so when going to the ChannelEdit SPA, you see the same "Channel not found" error page.

Reply:
I'll keep the Channel Permissions Error page in the code (even it's not used right now just in case)

Reply:
Thank you!